### PR TITLE
1770 la funded places

### PR DESCRIPTION
**Pull request by tariqhamid**

### CODE REVIEW

File: app/models/local_authority.rb [hunk 1]
```diff
@@ -15,4 +15,30 @@ class LocalAuthority < ResponsibleBody
   }
 
   validates :organisation_type, presence: true
+
+  def la_funded_place
+    schools.la_funded_place_establishment_type.first
```
Comment (nigel-lowry) [inline app/models/local_authority.rb:20]: Does this line work alright if there is no `first` object?

File: app/views/school/welcome_wizard/la_funded_devices_you_can_order.html.erb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      <p class="govuk-body">
+        These have temporary content filtering and mobile device management software installed, which you’ll
+        need to replace when they expire. You’ll need to reset these devices before you can apply new
+        settings, or download some types of software.
+      </p>
+      <h2 class="govuk-heading-s">
+        Google Chromebooks
+      </h2>
+      <p class="govuk-body">
+        You’ll need to provide us with the relevant school’s email domain for <span class="app-no-wrap">G
```
Comment (nigel-lowry) [inline app/views/school/welcome_wizard/la_funded_devices_you_can_order.html.erb:39]: You've got two of those annoying curly quotes in this line. This is something I know Macs insert to make the text look better. If you replace them with the single quote key on your keyboard, then we'll be able to search for this text in the future, otherwise the quote won't be matched (there's no curly quote key on your keyboard)

File: app/views/school/welcome_wizard/la_funded_place_allocation.html.erb [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            <%= @rb_name %> has been allocated <span class="app-no-wrap">
+                    <%= @allocation %> laptops
+                </span>
+          </h1>
+          <p class="govuk-body">
+            Your allocation estimate is based on the data <%= @rb_name %> submitted in the latest
+                published alternative provision census (2020).
+          </p>
+          <p class="govuk-body">
+            We’ve allocated one laptop for each pupil in years 3 to 11 that <%= @rb_name %> reported as
```
Comment (nigel-lowry) [inline app/views/school/welcome_wizard/la_funded_place_allocation.html.erb:18]: Another one of those pesky curly quotes

File: app/views/school/welcome_wizard/la_funded_devices_you_can_order.html.erb (hunk 2 above)
Comment (tonyheadford) [reply app/views/school/welcome_wizard/la_funded_devices_you_can_order.html.erb:39]: No, content needs to use the 6 and 9 style quotes

File: app/models/school.rb [hunk 4]
```diff
@@ -73,6 +74,10 @@ def is_further_education?
     type == 'FurtherEducationSchool'
   end
 
+  def is_la_funded_place?
```
Comment (nigel-lowry) [inline app/models/school.rb:77]: As mentioned on the call, if we use the more standard Ruby method predicate method name `la_funded_place?` then in our specs we can use `expect(x).to be_an_la_funded_place`

File: app/models/user.rb [hunk 5]
```diff
@@ -89,6 +89,10 @@ def is_school_user?
     school.present?
   end
 
+  def is_la_funded_user?
```
Comment (nigel-lowry) [inline app/models/user.rb:92]: Again we get some RSpec syntactic sugar if we replace `is_la_funded_user?` with `la_funded_user?`

File: app/models/user.rb [hunk 6]
```diff
@@ -89,6 +89,10 @@ def is_school_user?
...
+    schools.la_funded_place.any?
+  end
+
   def has_multiple_schools?
     schools.size > 1
```
Comment (nigel-lowry) [inline app/models/user.rb:97]: Not part of this commit, but this could be `schools.size.many?`

File: app/policies/la_funded_place_policy.rb [hunk 7]
```diff
@@ -0,0 +1,2 @@
+class LaFundedPlacePolicy < SchoolPolicy
```
Comment (nigel-lowry) [inline app/policies/la_funded_place_policy.rb:1]: Is this used?

File: app/views/school/welcome_wizard/la_funded_place_allocation.html.erb (hunk 3 above)
Comment (tonyheadford) [reply app/views/school/welcome_wizard/la_funded_place_allocation.html.erb:18]: See above

File: app/views/interstitials/_la_funded_user.html.erb [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+</h1>
+<p class="govuk-body">
+  Laptops are available for state-funded pupils that Newham funds to attend independent special schools or independent alternative provision.
+</p>
+<p class="govuk-body">
+  Use this service to:
+</p>
+<ul class="govuk-list govuk-list--bullet">
+  <li>
+    check you’ve been allocated the correct number of laptops
```
Comment (nigel-lowry) [inline app/views/interstitials/_la_funded_user.html.erb:16]: A pesky curly quote has crept in (probably from pasting in the text)

File: app/views/school/welcome_wizard/la_funded_devices_you_can_order.html.erb [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+      <p class="govuk-body">
+        Depending on availability, you can choose from:
+      </p>
+
+      <h2 class="govuk-heading-s">
+        Standard Windows laptops
+      </h2>
+      <p class="govuk-body">
+        You’ll be responsible for arranging and maintaining safeguarding measures, like content filtering.
```
Comment (nigel-lowry) [inline app/views/school/welcome_wizard/la_funded_devices_you_can_order.html.erb:24]: Curly quote

File: app/views/school/welcome_wizard/la_funded_devices_you_can_order.html.erb [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+      </p>
+      <h2 class="govuk-heading-s">
+        DfE Restricted Windows laptops
+      </h2>
+
+      <p class="govuk-body">
+        These have temporary content filtering and mobile device management software installed, which you’ll
```
Comment (nigel-lowry) [inline app/views/school/welcome_wizard/la_funded_devices_you_can_order.html.erb:31]: Curly quote

File: app/views/school/welcome_wizard/la_funded_devices_you_can_order.html.erb [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+        need to replace when they expire. You’ll need to reset these devices before you can apply new
```
Comment (nigel-lowry) [inline app/views/school/welcome_wizard/la_funded_devices_you_can_order.html.erb:32]: Curly quote

File: app/policies/la_funded_place_policy.rb (hunk 7 above)
Comment (tonyheadford) [reply app/policies/la_funded_place_policy.rb:1]: Yes, it's magic used by `pundit`

**Comment by asmega** [comment]
Can you please fill in the description of the pull request to provide the reviewer with some guidance. See https://github.com/alphagov/styleguides/blob/master/pull-requests.md for some reference and other pull requests raised in this repository

File: app/views/interstitials/_la_funded_user.html.erb [hunk 12]
```diff
@@ -0,0 +1,25 @@
+
```
Comment (asmega) [inline app/views/interstitials/_la_funded_user.html.erb:1]: can we fix the whitespace and indentation of this file

File: app/views/school/welcome_wizard/la_funded_place_allocation.html.erb [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+                school meals.
+          </p>
+
+          <p class="govuk-body">
+            You can request internet access for these pupils if they need it too.
+          </p>
+
+          <details class="govuk-details" data-module="govuk-details">
+            <summary class="govuk-details__summary">
+                <span class="govuk-details__summary-text">
```
Comment (asmega) [inline app/views/school/welcome_wizard/la_funded_place_allocation.html.erb:29]: only one level of indentation needed

File: spec/services/interstitial_picker_spec.rb [hunk 14]
```diff
@@ -0,0 +1,15 @@
+require 'rails_helper'
+
+RSpec.describe InterstitialPicker, type: :model do
```
Comment (asmega) [inline spec/services/interstitial_picker_spec.rb:3]: can we remove `type: model` as it not technically a activerecord model

File: app/views/school/welcome_wizard/la_funded_place_allocation.html.erb [hunk 15]
```diff
@@ -0,0 +1,62 @@
...
+              <p class="govuk-body">
+                Email this information to <a href="">COVID.TECHNOLOGY@education.gov.uk</a>.
+              </p>
+            </div>
+          </details>
+
+          <%= f.govuk_submit 'Continue' %>
+        <%- end %>
+      </div>
+    </div>
```
Comment (asmega) [inline app/views/school/welcome_wizard/la_funded_place_allocation.html.erb:62]: indentation has gone wrong somewhere as there should be no indentation on this line

File: app/views/school/welcome_wizard/la_funded_devices_you_can_order.html.erb [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+
+      <h2 class="govuk-heading-s">
+        Standard Windows laptops
+      </h2>
+      <p class="govuk-body">
+        You'll be responsible for arranging and maintaining safeguarding measures, like content filtering.
+      </p>
+      <h2 class="govuk-heading-s">
+        DfE Restricted Windows laptops
+      </h2>
```
Comment (asmega) [inline app/views/school/welcome_wizard/la_funded_devices_you_can_order.html.erb:28]: This hard coded list needs to be swapped for the partial see https://github.com/DFE-Digital/get-help-with-tech/blob/main/app/views/school/welcome_wizard/devices_you_can_order.html.erb#L15

Otherwise updates there will not be reflected on this page

File: app/views/interstitials/_la_funded_user.html.erb [hunk 17]
```diff
@@ -0,0 +1,20 @@
...
+</h1>
+<p class="govuk-body">
+  Laptops are available for state-funded pupils that Newham funds to attend independent special schools or independent alternative provision.
+</p>
+<p class="govuk-body">
+  Use this service to:
+</p>
+<ul class="govuk-list govuk-list--bullet">
+  <li>
+    check you've been allocated the correct number of laptops
```
Comment (asmega) [inline app/views/interstitials/_la_funded_user.html.erb:12]: need to use one of those smart apostrophes

File: app/views/school/welcome_wizard/la_funded_place_allocation.html.erb [hunk 18]
```diff
@@ -0,0 +1,62 @@
...
+            <%= @rb_name %> has been allocated <span class="app-no-wrap">
+                    <%= @allocation %> laptops
+                </span>
+          </h1>
+          <p class="govuk-body">
+            Your allocation estimate is based on the data <%= @rb_name %> submitted in the latest
+                published alternative provision census (2020).
+          </p>
+          <p class="govuk-body">
+            We've allocated one laptop for each pupil in years 3 to 11 that <%= @rb_name %> reported as
```
Comment (asmega) [inline app/views/school/welcome_wizard/la_funded_place_allocation.html.erb:18]: need to use one of those smart apostrophes

File: spec/services/interstitial_picker_spec.rb [hunk 19]
```diff
@@ -0,0 +1,15 @@
...
+  describe 'calling interstitial picker for la funded place' do
+    let(:school) { create :la_funded_place }
+    let(:user) { create :user, schools: [school] }
+    let(:service) { described_class.new(user: user) }
+
+    context 'when a standard device allocation does not exist' do
```
Comment (asmega) [inline spec/services/interstitial_picker_spec.rb:9]: this context looks to be superfluous and can be deleted

File: spec/services/interstitial_picker_spec.rb [hunk 20]
```diff
@@ -0,0 +1,15 @@
...
+      it 'creates a new device allocation record' do
```
Comment (asmega) [inline spec/services/interstitial_picker_spec.rb:10]: the name of this test does not correctly describe what it is asserting

File: app/views/school/welcome_wizard/la_funded_devices_you_can_order.html.erb [hunk 21]
```diff
@@ -0,0 +1,47 @@
+<div class="govuk-grid-row">
```
Comment (asmega) [inline app/views/school/welcome_wizard/la_funded_devices_you_can_order.html.erb:1]: this page will be missing a title see https://github.com/DFE-Digital/get-help-with-tech/blob/main/app/views/school/welcome_wizard/allocation.html.erb#L1

File: app/views/interstitials/_la_funded_user.html.erb [hunk 22]
```diff
@@ -0,0 +1,20 @@
+<h1 class="govuk-heading-xl">
+  Get laptops and internet access for <span class="app-no-wrap">state-funded</span> pupils at independent settings
+</h1>
+<p class="govuk-body">
+  Laptops are available for state-funded pupils that Newham funds to attend independent special schools or independent alternative provision.
```
Comment (tonyheadford) [inline app/views/interstitials/_la_funded_user.html.erb:5]: Hard-coded `Newham` here needs to be dynamic

**Review by tonyheadford — CHANGES_REQUESTED** [review]

File: app/views/school/welcome_wizard/la_funded_devices_you_can_order.html.erb [hunk 23]
```diff
@@ -0,0 +1,47 @@
...
+        settings, or download some types of software.
+      </p>
+      <h2 class="govuk-heading-s">
+        Google Chromebooks
+      </h2>
+      <p class="govuk-body">
+        You’ll need to provide us with the relevant school’s email domain for <span class="app-no-wrap">G
+            Suite for Education</span>, and a recovery email address.
+      </p>
+      <p class="govuk-body"><a href="/devices/device-specification">See detailed laptop specifications</a></p>
```
Comment (asmega) [inline app/views/school/welcome_wizard/la_funded_devices_you_can_order.html.erb:42]: this link needs to be a `govuk_link_to` otherwise the styling is wrong. search the codebase for `govuk_link_to` for examples

File: app/views/school/welcome_wizard/la_funded_devices_you_can_order.html.erb [hunk 24]
```diff
@@ -0,0 +1,50 @@
...
+        settings, or download some types of software.
+      </p>
+      <h2 class="govuk-heading-s">
+        Google Chromebooks
+      </h2>
+      <p class="govuk-body">
+        You’ll need to provide us with the relevant school’s email domain for <span class="app-no-wrap">G
+            Suite for Education</span>, and a recovery email address.
+      </p>
+      <p class="govuk_link_to"><a href="/devices/device-specification">See detailed laptop specifications</a></p>
```
Comment (jamesrose) [inline app/views/school/welcome_wizard/la_funded_devices_you_can_order.html.erb:45]: The `p` tag should be `govuk-body` and the link should be rendered with the method `govuk_link_to`.

**Review by jamesrose — CHANGES_REQUESTED** [review]

File: config/locales/en.yml [hunk 25]
```diff
@@ -253,6 +253,8 @@ en:
           choice: Tell us whether you will need Chromebooks
       what_happens_next:
         title: What happens next
+      range_of_devices:
```
Comment (asmega) [inline config/locales/en.yml:256]: this is no longer needed

**Comment by asmega** [comment]
can we also squash the commits. the only merge commits in a repository should be those of pull requests
